hocon/deployment/src/main/java/io/quarkus/config/hocon/deployment/ConfigHoconProcessor.java

```diff
@@ -0,0 +1,67 @@
+package io.quarkus.config.hocon.deployment;
+
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+
+import io.quarkus.config.hocon.runtime.ApplicationHoconConfigSourceLoader;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.AdditionalBootstrapConfigSourceProviderBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.HotDeploymentWatchedFileBuildItem;
+import io.quarkus.deployment.builditem.StaticInitConfigSourceProviderBuildItem;
+import io.smallrye.config.SmallRyeConfig;
+
+public final class ConfigHoconProcessor {
+
+    private static final String FEATURE = "hocon-config";
+
+    @BuildStep
+    public void feature(BuildProducer<FeatureBuildItem> feature) {
+        feature.produce(new FeatureBuildItem(FEATURE));
+    }
+
+    @BuildStep
+    public void bootstrap(
+            BuildProducer<AdditionalBootstrapConfigSourceProviderBuildItem> additionalBootstrapConfigSourceProvider,
+            BuildProducer<StaticInitConfigSourceProviderBuildItem> staticInitConfigSourceProvider) {
+        additionalBootstrapConfigSourceProvider.produce(new AdditionalBootstrapConfigSourceProviderBuildItem(
+                ApplicationHoconConfigSourceLoader.InFileSystem.class.getName()));
+        additionalBootstrapConfigSourceProvider.produce(new AdditionalBootstrapConfigSourceProviderBuildItem(
+                ApplicationHoconConfigSourceLoader.InClassPath.class.getName()));
+        staticInitConfigSourceProvider.produce(new StaticInitConfigSourceProviderBuildItem(
+                ApplicationHoconConfigSourceLoader.InFileSystem.class.getName()));
+        staticInitConfigSourceProvider.produce(new StaticInitConfigSourceProviderBuildItem(
+                ApplicationHoconConfigSourceLoader.InClassPath.class.getName()));
```

Comment:
I know we have this in the YAML source, but I recently submitted a PR to use our builder step. I plan to deprecate all of these. Check here:
https://github.com/quarkusio/quarkus/pull/30356/files#diff-6d620b5b84d3f7ea9c637cc33bcc75219b91df94f4d4ec872b410837607c451f
https://github.com/quarkusio/quarkus/pull/30356/files#diff-0f949a92f29bc67d8932bf3f9b014337db84d24b3e5bd774ff5926602689ae4f